core/controllers/suggestion.py

```diff
@@ -50,6 +50,38 @@ def _require_valid_suggestion_and_target_types(target_type, suggestion_type):
             'Invalid suggestion_type: %s' % suggestion_type)
 
 
+def _render_suggestions(self, target_type, suggestions):
+    """Renders retrieved suggestions for a handler."""
+    target_ids = set([s.target_id for s in suggestions])
+    if target_type == suggestion_models.TARGET_TYPE_EXPLORATION:
+        target_ids_to_opportunities = (
+            opportunity_services
+            .get_exploration_opportunity_summaries_by_ids(
+                list(target_ids)))
+        _render_suggestions_and_opportunities(
+            self, suggestions, target_ids_to_opportunities)
+    elif target_type == suggestion_models.TARGET_TYPE_SKILL:
+        target_ids_to_opportunities = (
+            opportunity_services.get_skill_opportunities_by_ids(
+                list(target_ids)))
+        _render_suggestions_and_opportunities(
+            self, suggestions, target_ids_to_opportunities)
+    else:
+        self.render_json({})
+
+
+def _render_suggestions_and_opportunities(
+        self, suggestions, target_ids_to_opportunities):
+    """Renders suggestions and opportunities for a handler."""
+    self.render_json({
```

Comment:
Maybe this function just returns the dict that needs to be rendered, and then in the controller you can do self.render?

Reply:
I created a parent class with a helper method as suggested above.